crypto/dynamic_loading_test.c

```diff
@@ -0,0 +1,87 @@
+#include <dlfcn.h>
```

Comment:
Can you add documentation in the header to describe what this is actually testing and what was failing previously?